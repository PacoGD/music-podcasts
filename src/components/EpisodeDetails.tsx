interface Episode {
  title: string;
  description: string;
  audioUrl: string;
}

interface EpisodeDetailsProps {
  episode: Episode;
}

function EpisodeDetails({ episode }: EpisodeDetailsProps) {
  return (
    <div>
      <h2>{episode.title}</h2>
      <p>Descripci√≥n: {episode.description}</p>
      <audio controls>
        <source src={episode.audioUrl} type="audio/mpeg" />
        Tu navegador no soporta el elemento de audio.
      </audio>
    </div>
  );
}

export default EpisodeDetails;
