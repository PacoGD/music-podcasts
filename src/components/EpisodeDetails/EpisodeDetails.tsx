import { useAppSelector } from "../../hooks/store";
import { Episode } from "../../types/types";
import styles from "./style.module.css";

function formatTitle(episode: Episode | null) {
  return episode?.artistName !== undefined
    ? `${episode?.trackName} - ${episode?.artistName}`
    : episode?.trackName;
}

function EpisodeDetails() {
  const episode: Episode | null = useAppSelector(
    (state) => state.episodes.episode
  );
  return (
    <div className={styles.container}>
      <p className={styles.title}>{formatTitle(episode)}</p>
      <p className={styles.description}>{episode?.description}</p>
      <audio controls>
        <source
          className={styles.sourceAudio}
          src={episode?.episodeUrl}
          type="audio/mpeg"
        />
      </audio>
    </div>
  );
}

export default EpisodeDetails;
