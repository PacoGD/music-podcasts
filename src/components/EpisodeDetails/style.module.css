.container {
  width: 400px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.25);
  margin: 15px 0 30px 0;
  padding: 15px;
}

.title {
  font-size: 14px;
  font-weight: 700;
  margin-left: 7px;
  text-align: left;
}

.description {
  font-size: 10px;
  color: #000;
  text-align: left;
  font-style: italic;
  margin: 0 5px 0 5px;
  overflow: hidden;
}

audio {
  height: 20px;
  color: #000;
  width: 100%;
  margin: 30px 5px 0 5px;
}
