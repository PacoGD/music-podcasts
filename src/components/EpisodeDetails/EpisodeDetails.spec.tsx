import { render } from "@testing-library/react";
import EpisodeDetails from "./EpisodeDetails";

jest.mock("../../hooks/store", () => ({
  useAppSelector: jest.fn(() => {
    ("");
  }),
}));

describe("EpisodeDetails component", () => {
  test("snapshot", () => {
    const { container } = render(<EpisodeDetails />);
    expect(container).toMatchSnapshot();
  });
});
