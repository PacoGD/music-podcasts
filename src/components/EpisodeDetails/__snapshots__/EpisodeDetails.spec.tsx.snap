// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeDetails component snapshot 1`] = `
<div>
  <div>
    <p />
    <div />
    <audio
      controls=""
    >
      <source
        type="audio/mpeg"
      />
    </audio>
  </div>
</div>
`;
